src/CubicBezierLength.ts

```diff
@@ -0,0 +1,780 @@
+    0.0522933351526832859403120512732112561121,
+    0.0337749015848141547933022468659129013491,
+    0.0337749015848141547933022468659129013491,
+    0.0146279952982722006849910980471854451902,
+    0.0146279952982722006849910980471854451902
+  ],
+  [
+    0.1336545721861061753514571105458443385831,
+    0.132462039404696617371642464703316925805,
+    0.132462039404696617371642464703316925805,
+    0.1289057221880821499785953393997936532597,
+    0.1289057221880821499785953393997936532597,
+    0.1230490843067295304675784006720096548158,
+    0.1230490843067295304675784006720096548158,
+    0.1149966402224113649416435129339613014914,
+    0.1149966402224113649416435129339613014914,
+    0.1048920914645414100740861850147438548584,
+    0.1048920914645414100740861850147438548584,
+    0.0929157660600351474770186173697646486034,
+    0.0929157660600351474770186173697646486034,
+    0.0792814117767189549228925247420432269137,
+    0.0792814117767189549228925247420432269137,
+    0.0642324214085258521271696151589109980391,
+    0.0642324214085258521271696151589109980391,
+    0.0480376717310846685716410716320339965612,
+    0.0480376717310846685716410716320339965612,
+    0.0309880058569794443106942196418845053837,
+    0.0309880058569794443106942196418845053837,
+    0.0134118594871417720813094934586150649766,
+    0.0134118594871417720813094934586150649766
+  ],
+  [
+    0.1279381953467521569740561652246953718517,
+    0.1279381953467521569740561652246953718517,
+    0.1258374563468282961213753825111836887264,
+    0.1258374563468282961213753825111836887264,
+    0.121670472927803391204463153476262425607,
+    0.121670472927803391204463153476262425607,
+    0.1155056680537256013533444839067835598622,
+    0.1155056680537256013533444839067835598622,
+    0.1074442701159656347825773424466062227946,
+    0.1074442701159656347825773424466062227946,
+    0.0976186521041138882698806644642471544279,
+    0.0976186521041138882698806644642471544279,
+    0.086190161531953275917185202983742667185,
+    0.086190161531953275917185202983742667185,
+    0.0733464814110803057340336152531165181193,
+    0.0733464814110803057340336152531165181193,
+    0.0592985849154367807463677585001085845412,
+    0.0592985849154367807463677585001085845412,
+    0.0442774388174198061686027482113382288593,
+    0.0442774388174198061686027482113382288593,
+    0.0285313886289336631813078159518782864491,
+    0.0285313886289336631813078159518782864491,
+    0.0123412297999871995468056670700372915759,
+    0.0123412297999871995468056670700372915759
+  ]
+];
+
+// LUT for binomial coefficient arrays per curve order 'n'
+const binomialCoefficients = [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1]];
+
+// Look up what the binomial coefficient is for pair {n,k}
+const binomials = (n: number, k: number) => binomialCoefficients[n][k];
+
+/**
+ * Compute the curve derivative (hodograph) at t.
+ */
+const getDerivative = (derivative: number, t: number, vs: number[]): number => {
```

Comment:
It's kinda magical for me :)
By the way, the code looks good but could we add some example on snack + inner in package later? As it is very advanced code and I'm pretty sure without it there might appear issues with questions `How` and `Why`.